### DISCUSSION TITLE
PUT method  400 bad request error

### DISCUSSION THREAD

**Discussion by alvinluo666** [laravel/framework#38742] [Q&A]
- Laravel Version: 5.1.*
- PHP Version: 7.1
- Database Driver & Version: MYSQL 5.7

### Description:
I couldn't use _method:put in form data to sent put request.
Everytime I add **_method:put** to form data , It generate 400 browser could not understand your request.
When i remove **_method:put** , It works as normal but it go with POST method.

**Comment by Jubeki** [comment]
This seems like a question for one of the support channels.
You find all support channels here: https://laravel.com/docs/8.x/contributions#support-questions

What you need to Put as an Input into the Form:
```html
<input type="hidden" name="_method" value="put" />
```

If you want to make a Request with Ajax, then you would put it there.
I don't know where you added `_method:put` because you didn't provide any example code.
Also did you use a "PUT" or only a "POST" route?

**Comment by Jubeki** [comment]
Also please check the currently supported Laravel Versions: https://laravel.com/docs/8.x/releases#support-policy

And upgrade if possible.

**Comment by alvinluo666** [comment]
> This seems like a question for one of the support channels.
> You find all support channels here: https://laravel.com/docs/8.x/contributions#support-questions
> 
> What you need to Put as an Input into the Form:
> 
> ```
> <input type="hidden" name="_method" value="put" />
> ```
> 
> If you want to make a Request with Ajax, then you would put it there.
> I don't know where you added `_method:put` because you didn't provide any example code.
> Also did you use a "PUT" or only a "POST" route?

Yes i have included _method field in every form request
```
 <form data-parsley-validate novalidate class="form-horizontal" action="{{ admin_url('single/'.$single->user_id) }}" method="post">
              {!! csrf_field() !!}
              {!! method_field('PUT') !!}
```

**Comment by daniel-koertvelyessy** [comment]
You can also use the  `@method('put')` and `@csrf` directives within you blade template.

**Comment by Jubeki** [comment]
@daniel-koertvelyessy I am not sure if this was possible in Laravel 5.1 https://laravel.com/docs/5.1/routing#form-method-spoofing

**Comment by alvinluo666** [comment]
> @daniel-koertvelyessy I am not sure if this was possible in Laravel 5.1 https://laravel.com/docs/5.1/routing#form-method-spoofing

Is there a chance that laravel generate 400 bad request error ? Sir

**Comment by Jubeki** [comment]
I don't know for Version 5.1 you need to share a little bit of code (routes, controllers, blade, ...)

Also I only joint after version 5.6.

I recommend upgrading to a newer version for more features and for security reasons.
Maybe you should checkout https://laravelshift.com for upgrading if you don't have the time.
